modules/layers/src/solid-polygon-layer/polygon-tesselator.js

```diff
@@ -93,73 +61,89 @@ export class PolygonTesselator {
     return this.attributes.vertexValid;
   }
 
-  elevations({key = 'elevations', getElevation = x => 100} = {}) {
-    const {attributes, polygons, pointCount} = this;
-    const values = updateElevations({cache: attributes[key], polygons, pointCount, getElevation});
-    attributes[key] = values;
-    return values;
+  elevations({target, getElevation}) {
+    return this._updateAttribute({
+      target,
+      size: 1,
+      getValue: object => [getElevation(object)]
+    });
   }
 
-  colors({key = 'colors', getColor = x => DEFAULT_COLOR} = {}) {
-    const {attributes, polygons, pointCount} = this;
-    const values = updateColors({cache: attributes[key], polygons, pointCount, getColor});
-    attributes[key] = values;
-    return values;
+  colors({target, getColor}) {
+    return this._updateAttribute({
+      target,
+      size: 4,
+      getValue: object => {
+        const color = getColor(object);
+        if (isNaN(color[3])) {
+          color[3] = 255;
+        }
+        return color;
+      }
+    });
   }
 
-  pickingColors() {
-    return this.attributes.pickingColors;
+  pickingColors({target, getPickingColor}) {
+    return this._updateAttribute({
+      target,
+      size: 3,
+      getValue: (object, index) => getPickingColor(index)
+    });
   }
-}
 
-// Count number of points in a list of complex polygons
-function getPointCount(polygons) {
-  return polygons.reduce((points, polygon) => points + Polygon.getVertexCount(polygon), 0);
-}
+  /* Utils */
+  countInstancesInGeometry(polygon) {
```

Comment:
Nit: This is not a "util", this is the subclass' redefinition of the virtual functions/callbacks defined by the base class. I would keep these together here after the and then list the actual utils (which should be private) last.